webapp/channels/src/actions/notification_actions.jsx

```diff
@@ -190,17 +194,27 @@ export function sendDesktopNotification(post, msgProps) {
         }
         notify = notify || !state.views.browser.focused;
 
-        const soundName = getNotificationSoundFromChannelMemberAndUser(member, user);
+        let soundName = getNotificationSoundFromChannelMemberAndUser(member, user);
 
-        if (notify) {
-            const updatedState = getState();
-            let url = getChannelURL(updatedState, channel, teamId);
+        const updatedState = getState();
+        let url = getChannelURL(updatedState, channel, teamId);
 
-            if (isCrtReply) {
-                url = getPermalinkURL(updatedState, teamId, post.id);
-            }
+        if (isCrtReply) {
+            url = getPermalinkURL(updatedState, teamId, post.id);
+        }
 
-            dispatch(notifyMe(title, body, channel, teamId, !sound, soundName, url));
+        // Allow plugins to change the notification, or re-enable a notification
+        const hookResult = await dispatch(runDesktopNotificationHooks(post, msgProps, channel, teamId, title, body, !sound, soundName, url, notify));
+        if (hookResult.error) {
+            dispatch(logError(hookResult.error));
+            return;
+        }
+
+        let silent = false;
+        ({title, body, silent, soundName, url, notify} = hookResult.args);
```

Comment:
This is a little scary :p

Reply:
Which part? This is js, scary is normal. :p